api/namex/models/request.py

```diff
@@ -623,6 +626,20 @@ def include_designations_in_name(cls, special_characters_name, any_designation_l
         return full_name
 
 
+@event.listens_for(Request, 'after_insert')
+@event.listens_for(Request, 'after_update')
+def on_insert_or_update_nr(mapper, connection, request):
+    """Send a new cloud event message on changes for stateCd in the Request model.
+       
+       Temporary NRs (nrNum starting with 'NR L') are discarded.
+    """
+    if (len(get_history(request, 'nrNum').added) or 
```

Comment:
start with the cheapest failing operation
not request.nrNum.startswith('NR L')
and get the history once
```suggestion
    if not request.nrNum.startswith('NR L') and (history := get_history(request, 'stateCd')):
```

we don't want to send messages unless it's a state change, maybe just those in State. RELEASE_STATES

Reply:
Yes, you're right. Thanks. I didn't changed to what you suggested but I got your point and improved it.